Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorSystemComponent.cpp

```diff
@@ -231,79 +231,82 @@ namespace Multiplayer
-                AZ_Assert(editorNetworkInterface, "MP Editor Network Interface was unregistered before Editor could connect.")
                 
-                m_editorConnId = editorNetworkInterface->Connect(AzNetworking::IpAddress(remoteAddress.c_str(), editorsv_port, AzNetworking::ProtocolType::Tcp));
-
-                if (m_editorConnId == AzNetworking::InvalidConnectionId)
-                {
-                    AZ_Warning(
-                        "MultiplayerEditor", false,
-                        "Editor multiplayer game-mode failed! Could not connect to an editor-server. editorsv_launch is false so we're assuming you're running your own editor-server at editorsv_serveraddr(%s) on editorsv_port(%i). "
-                        "Either set editorsv_launch=true so the editor launches an editor-server for you, or launch your own editor-server by hand before entering game-mode. Remember editor-servers must use editorsv_isDedicated=true.",
-                        remoteAddress.c_str(),
-                        static_cast<uint16_t>(editorsv_port))
-                    return;
-                }
-
-                SendEditorServerLevelDataPacket(editorNetworkInterface->GetConnectionSet().GetConnection(m_editorConnId));
+            // Begin listening for MPEditor packets before we launch the editor-server.
+            // The editor-server will send us (the editor) an "EditorServerReadyForLevelData" packet to let us know it's ready to receive data.
+            INetworkInterface* editorNetworkInterface =
+                AZ::Interface<INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpEditorInterfaceName));
+            AZ_Assert(editorNetworkInterface, "MP Editor Network Interface was unregistered before Editor could connect.");
+            editorNetworkInterface->Listen(editorsv_port);
+
+            // Launch the editor-server
+            m_serverProcess = LaunchEditorServer();
+        }
+        else
+        {
+            // Editorsv_launch=false, so we're expecting an editor-server already exists.
+            // Connect to the editor-server and then send the EditorServerLevelData packet.
+            INetworkInterface* editorNetworkInterface = AZ::Interface<INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpEditorInterfaceName));
+            AZ_Assert(editorNetworkInterface, "MP Editor Network Interface was unregistered before Editor could connect.")
+                
+            m_editorConnId = editorNetworkInterface->Connect(AzNetworking::IpAddress(remoteAddress.c_str(), editorsv_port, AzNetworking::ProtocolType::Tcp));
+
+            if (m_editorConnId == AzNetworking::InvalidConnectionId)
+            {
+                AZ_Warning(
+                    "MultiplayerEditor", false,
+                    "Editor multiplayer game-mode failed! Could not connect to an editor-server. editorsv_launch is false so we're assuming you're running your own editor-server at editorsv_serveraddr(%s) on editorsv_port(%i). "
```

Comment:
".. Could connect to a server at editorsv_serveraddr(%s) on editorsv_port(%i). Check server is active or use editorsv_launch to auto-launch a server to connect to. "

Reply:
yep, changed!